### PR TITLE
Ability for user to unjoin from an existing hack team

### PR DESCRIPTION
**Pull request by HuangStanley050**
* new lambda unjoinHack
* same idea as joinHack, take a hackId and userId from post body
* conditions to allow users to join a hack

### CODE REVIEW

File: services/hack/joinHack.js [hunk 1]
```diff
@@ -17,31 +17,30 @@ export const join = async (event, context) => {
...
-    //     }
-    //   });
-    // }
+    //console.log(res);
+
+    if (res.status === "Submitted") throw new Error("Unable to join");
+
+    if (res.status === "Canceled") throw new Error("Hack is canceld");
+
+    if (res.status === "Team Closed") throw new Error("Hack is closed");
```
Comment (josephgarner) [inline services/hack/joinHack.js]: Just a note on this, We should look at disabling the join button with these statuses

Comment (josephgarner) [reply services/hack/joinHack.js]: one other thing to clean this up a little, something that Jo showed me.
you could try
`if(["Submitted", "Canceled", "Team Closed"].includes(res.status)) throw new Error("Unable to join hack due to status")`
just cleans those three lines up into one

**Review by josephgarner — APPROVED** [review]
Just a comment on the statuses all looks good though

File: services/hack/joinHack.js (hunk 1 above)
Comment (HuangStanley050) [reply services/hack/joinHack.js]: We will change the logic a bit on the front end to disable the button, which shouldn't be a problem.

**Pull request merged**